sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py

```diff
@@ -413,7 +422,9 @@ def refresh(self, **kwargs) -> None:
             return
         try:
             with self._update_lock:
-                for (key, label), etag in self._refresh_on.items():
+                need_refresh = False
+                updated_sentinel_keys = dict(self._refresh_on)
+                for (key, label), etag in updated_sentinel_keys.items():
                     updated_sentinel = self._client.get_configuration_setting(
                         key=key, label=label, etag=etag, match_condition=MatchConditions.IfModified, **kwargs
                     )
```

Comment:
Just looked at the doc. It looks like it should throw an error. I'll at a check for this.

Reply:
@avanigupta, just tested it. It does fail. Should we 1, fail the refresh the value was there at the start and it isn't now. This current either results in refresh throwing an error, or the on_refresh_fail callback happening. 2, we consider this a change to the key and it is set to None as the etag, so adding it back will cause a refresh.


This issue also appears at startup if a listed key is not found. Result s a ResourceNotFoundError.

Minimally we should throw a new error to clarify that a WatchKey wasn't found.

Reply:
I am thinking a throw on startup and a warning while running would be a good idea. I would be interested to know how the dotnet provider handles such a situation though

Reply:
Updating to catch changes, but only on refresh. If deleted the etag will be set to null so any future update will cause a refresh if the key is created again.

Reply:
If WatchKey is not found, we should ignore it during startup. During refresh, if WatchKey is not found, we need to check if this key was present earlier to tell the difference between deleted sentinel key vs non-existent sentinel key. If a sentinel key is deleted, trigger refresh. Otherwise, just like startup, keep ignoring non-existent sentinel keys. That's what we do in .NET provider too.